BEIMA.Backend.Test/MongoService/DeviceTest.cs

```diff
@@ -105,29 +74,29 @@ public void DeviceNotInstantiated_InstantiateUsingFullConstructor_DeviceInstanti
             Assert.That(device.Notes, Is.EqualTo(validNotes));
 
             var lastModified = device.LastModified;
-            Assert.That((DateTime)lastModified.GetElement("date").Value, Is.EqualTo(validDate).Within(5).Seconds);
-            Assert.That((string)lastModified.GetElement("user").Value, Is.EqualTo(validUser));
+            Assert.That(lastModified.Date, Is.EqualTo(validDate).Within(5).Seconds);
+            Assert.That(lastModified.User, Is.EqualTo(validUser));
 
             var location = device.Location;
-            Assert.That((ObjectId)location.GetElement("buildingId").Value, Is.EqualTo(validBuildingId));
-            Assert.That((string)location.GetElement("notes").Value, Is.EqualTo(validLocationNotes));
-            Assert.That((string)location.GetElement("latitude").Value, Is.EqualTo(validLatitude));
-            Assert.That((string)location.GetElement("longitude").Value, Is.EqualTo(validLongitude));
+            Assert.That(location.BuildingId, Is.EqualTo(validBuildingId));
+            Assert.That(location.Notes, Is.EqualTo(validLocationNotes));
+            Assert.That(location.Latitude, Is.EqualTo(validLatitude));
+            Assert.That(location.Longitude, Is.EqualTo(validLongitude));
 
             var fields = device.Fields;
-            Assert.That((string)fields.GetElement(validKey1).Value, Is.EqualTo(validValue1));
-            Assert.That((bool)fields.GetElement(validKey2).Value, Is.EqualTo(validValue2));
-            Assert.That((int)fields.GetElement(validKey3).Value, Is.EqualTo(validValue3));
+            Assert.That(fields[validKey1], Is.EqualTo(validValue1));
+            Assert.That(fields[validKey2], Is.EqualTo(validValue2));
+            Assert.That(fields[validKey3], Is.EqualTo(validValue3));
 
             var files = device.Files;
-            Assert.That(files.Contains(file1), Is.True);
-            Assert.That(files.Contains(file2), Is.True);
-            Assert.That(files.Contains(file3), Is.True);
+            Assert.That(files.Any(file => file.FileUid == validFileUid1 && file.FileName == validFileName1), Is.True);
```

Comment:
```suggestion
            Assert.That(files.Single(file => file.FileUid == validFileUid1).FileName, Is.EqualTo(validFileName1));
```
I think I like doing it this way better, but that's just my personal preference.